Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/util/SecurityUtils.java

```diff
@@ -0,0 +1,245 @@
+                                     final String dataStr,
+                                     final byte[] signedMsg,
+                                     final Logger logger) {
+        notNullAssert(publicKey, "publicKey");
+        notNullAssert(dataStr, "dataStr");
+        notNullAssert(signedMsg, "signedMsg");
+        notNullAssert(logger, "logger");
+
+        boolean verified = false;
+
+        try {
+            final Signature signature = Signature.getInstance(SIGNATURE_ALGORITHM);
+
+            signature.initVerify(publicKey);
+
+            signature.update(dataStr.getBytes(StandardCharsets.UTF_8));
+
+            verified = signature.verify(signedMsg);
+        } catch (final Exception e) {
+            logger.error("\n\nException: {}", e.getMessage());
+
+            e.printStackTrace();
+        } finally {
+            return verified;
+        }
+    }
+
+    public static String symmetricEncrypt(final String strToEncrypt, final String encryptionSecret) {
+        notNullAssert(strToEncrypt, "strToEncrypt");
+        notNullAssert(encryptionSecret, "encryptionSecret");
+
+        try {
+            setKey(encryptionSecret);
+
+            final Cipher cipher = Cipher.getInstance(SYMMETRIC_CIPHER_FULL_ALG_SPEC);
+            cipher.init(Cipher.ENCRYPT_MODE, SecurityUtils.secretKey);
+
+            return Base64.getEncoder().encodeToString(cipher.doFinal(strToEncrypt.getBytes("UTF-8")));
+        } catch (final Exception e) {
+            System.out.println("Error while encrypting: " + e.toString());
+        }
+
+        return null;
+    }
+
+    public static String symmetricDecrypt(final String strToDecrypt, final String encryptionSecret) throws Exception {
+        notNullAssert(strToDecrypt, "strToDecrypt");
+        notNullAssert(encryptionSecret, "encryptionSecret");
+
+        try {
+            setKey(encryptionSecret);
+
+            final Cipher cipher = Cipher.getInstance(SYMMETRIC_CIPHER_FULL_ALG_SPEC);
+            cipher.init(Cipher.DECRYPT_MODE, SecurityUtils.secretKey);
+
+            return new String(cipher.doFinal(Base64.getDecoder().decode(strToDecrypt)));
+        } catch (final Exception e) {
+            System.out.println("Error while decrypting: " + e.toString());
+
+            throw e;
+        }
+    }
+
+    private static void setKey(final String myKey) {
+        MessageDigest sha;
+
+        try {
+            key = myKey.getBytes(CHARSET_ENCODING);
+            sha = MessageDigest.getInstance(DIGEST_ALGORITHM);
+            key = sha.digest(key);
+            key = Arrays.copyOf(key, 16);
+            secretKey = new SecretKeySpec(key, SYMMETRIC_CIPHER_BASE_ALG);
+        } catch (final NoSuchAlgorithmException noSuchAlgorithmException) {
+            noSuchAlgorithmException.printStackTrace();
+        } catch (final UnsupportedEncodingException unsupportedEncodingException) {
+            unsupportedEncodingException.printStackTrace();
```

Comment:
Don't log to `System.out`.

Reply:
Updated exception handling, via logger, per review comments [DONE].